src/wasm.h

```diff
@@ -1441,6 +1441,8 @@ class Module {
   std::map<Name, Import*> importsMap;
   std::map<Name, Export*> exportsMap;
   std::map<Name, Function*> functionsMap;
+  // Types of functions which are declared but not defined.
+  std::map<Name, FunctionType*> externTypesMap;
```

Comment:
It does need to exist somewhere; a functionType is just a type (it can have a name, but that' the name for the type, i.e. the asm.js-style signature). We still need to have a list of the external functions that are not part of the module but that the linker knows how to generate thunks for. You're right that it doesn't necessarily need to be on the module. It could be in the linker instead. Although in that case it would be weird to have the types in the module, and the functions in the linker; probably would put everything in the linker.